packages/android_intent/android/src/main/java/dev/flutter/plugins/androidintent/MethodCallHandlerImpl.java

```diff
@@ -0,0 +1,132 @@
+package dev.flutter.plugins.androidintent;
+
+import android.content.ComponentName;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.text.TextUtils;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import java.util.ArrayList;
+import java.util.Map;
+
+/** Forwards incoming {@link MethodCall}s to {@link IntentSender#send}. */
+public class MethodCallHandlerImpl implements MethodCallHandler {
+  private final IntentSender sender;
+
+  /**
+   * Uses the given {@code sender} for all incoming calls.
+   *
+   * <p>This assumes that the sender's context and activity state are managed elsewhere and
+   * correctly initialized before being sent here.
+   */
+  public MethodCallHandlerImpl(IntentSender sender) {
+    this.sender = sender;
+  }
+
+  /**
+   * Parses the incoming call and forwards it to the cached {@link IntentSender}.
+   *
+   * <p>Always calls {@code result#success}.
+   */
+  @Override
+  public void onMethodCall(MethodCall call, Result result) {
```

Comment:
I'm not sure if annotations are missing by accident, or on purpose, but FYI.

Reply:
Done.